src/interactivity/base/RemoteConsoleControl.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+
+#include "RemoteConsoleControl.hpp"
+
+#include "../../inc/HostSignals.hpp"
+
+using namespace Microsoft::Console::Interactivity;
+
+RemoteConsoleControl::RemoteConsoleControl(HANDLE signalPipe) :
+    _pipe{ signalPipe }
+{
+}
+
+#pragma region IConsoleControl Members
+
+[[nodiscard]] NTSTATUS RemoteConsoleControl::NotifyConsoleApplication(_In_ DWORD dwProcessId)
+{
+    DWORD dwWritten = 0;
+
+    unsigned short code = HOST_SIGNAL_NOTIFY_APP;
+    if (!WriteFile(_pipe.get(), &code, sizeof(code), &dwWritten, nullptr))
+    {
+        return NTSTATUS_FROM_WIN32(::GetLastError());
+    }
+
+    HOST_SIGNAL_NOTIFY_APP_DATA data;

```

Comment:
```suggestion
    HOST_SIGNAL_NOTIFY_APP_DATA data{};
```

even though you're going to initialize every member, it doesn't hurt to zero-initialize it too